packages/assets-controllers/src/NftController.ts

```diff
@@ -1027,47 +1021,29 @@ export class NftController extends BaseController<NftConfig, NftState> {
       throw rpcErrors.invalidParams('Invalid tokenId');
     }
 
-    // Check if the suggested NFT is already pending
-    // if (
     // TODO: check if the suggested NFT is already pending in approval controller state
+    // if (
+    //   this.approvalController.state.pendingApproval.find(({requestData: {asset: {address, tokenId}}}) => contractAddress === contractAddress && tokenId === tokenId)
     // ) {
     //   throw rpcErrors.internal('Suggested NFT already pending');
     // }
 
-    const errors = {
-      nftAlreadyWatchedError: false,
-      ownerFetchError: false,
-      wrongOwnerError: false,
-    };
-    // Check if the suggested NFT is already being watched by the selected account
-    if (
-      this.state.allNfts?.[accountAddress]?.[chainId]?.find(
-        ({ address: watchedAssetAddress, tokenId: watchedAssetTokenId }) => {
-          return (
-            watchedAssetAddress === contractAddress &&
-            watchedAssetTokenId === tokenId
-          );
-        },
-      )
-    ) {
-      errors.nftAlreadyWatchedError = true;
-      return errors;
-    }
-
+    // Check if the user owns the suggested NFT
     try {
       const isOwner = await this.isNftOwner(
         accountAddress,
         contractAddress,
         tokenId,
       );
       if (!isOwner) {
-        errors.wrongOwnerError = true;
+        throw rpcErrors.invalidInput(
+          'Suggested NFT is not owned by the selected account',
+        );
```

Comment:
@Gudahtt @shanejonas I've put this unable to verify owner error as a `invalidInput` code for now, since as Mark pointed out the dApp should not have sent a request for an asset the user does not own.